executor.go

```diff
@@ -0,0 +1,139 @@
+package goptions
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"path/filepath"
+	"reflect"
+)
+
+var (
+	//ErrNoCommand This should never happen.
+	ErrNoCommand = errors.New("No command specified")
+)
+
+//Executer used to allow auto execution of commands.
+type Executer interface {
+	Execute(args []string) error
+}
+
+// ParseAndExecute ...
```

Comment:
These comments are not really documenting anything. Would you mind augmenting them with an actual description or even proper examples??